cli/src/commands/git.rs

```diff
@@ -964,21 +965,18 @@ fn cmd_git_push(
     for (branch_name, update) in &branch_updates {
```

Comment:
@yuja, continuing the discussion from https://github.com/martinvonz/jj/pull/3522#discussion_r1615391352:

> I personally wouldn't mind adding two more match arms there, but I agree it can be addressed later. I think it's also good to split the last commit to new PR.
> 
> FWIW, I would use something like `{branch} (added|deleted|moved forward|...): {old} -> {new}` to keep the formatting code small. I think `jj git fetch` uses a similar formatting.

Do you prefer:

```
{branch} deleted: {old} -> (missing)
{branch} added:   (missing) -> {old}
```

or some more English:

```
{branch} added to {new}
{branch} deleted from {old}
{branch} moved forward: {old} -> {new}
```

to me, the latter seems a lot clearer.

Reply:
I don't have any preference over these. I would choose whatever that keeps the code simple.

FWIW, `jj git fetch` uses the following output, but I'm not a big fan of this either:
```
branch: a1@origin     [updated] tracked
branch: a2@origin     [updated] tracked
branch: b@origin      [updated] tracked
branch: trunk2@origin [new] tracked
```
(I personally don't buy much about column alignment.)

Reply:
Yuya, I'm sorry, but I couldn't intuit a way to make the code simpler  by changing the messages. Perhaps I'm missing something obvious. So, I kept the messages the same (I actually like the English sentences, though I'm not opposed to changing them).

I refactored things a bit in a separate commit. I think the code got a bit better, let me know if you agree or if you have ideas on how to further refactor.

If you dislike both the older and new version of the PR, I care mostly about changing the "Force..." message to something more accurate before the release. I'd change it to just "Move..." for simplicity, but then the new version of `jj` would output less information than the older version. I could replace this PR with a commit changing "Force..." to always say "Move sideways/backwards" or just "Move sideways", not adding any complexity to what it was, if you prefer. You can then refactor it further if you see a way to improve things.

Reply:
What I want to avoid is constructing a sentence as `"{verb} something"` or `"do {adjective} something"`. The `"do something [{auxiliary_info}]"` form is okay, and if that help simplify the formatting, it can be used.

FWIW, I personally don't mind if the message formatting code were something like:
```rust
            (Some(old_target), Some(new_target), Forward) => {
                writeln!(
                    ui.status(),
                    "  Move branch {branch_name} forward from {old} to {new}",
                    ..
            }
            (Some(old_target), Some(new_target), Backward) => {
                writeln!(
                    ui.status(),
                    "  Move branch {branch_name} backward from {old} to {new}",
                    ..
            }
            ...
```
As I said, we already have three or four match arms, so I don't think it becomes messier by adding a few more.

Reply:
OK, I changed this PR to just make this change you requested. I'll put additional refactors in a separate PR.

> What I want to avoid is constructing a sentence as "{verb} something" or "do {adjective} something". 

You clearly feel much stronger about this than I do. I agree that overdoing this is bad, and I think you have caught me overdoing it a few times in the past. I may learn something later that will change my mind, but I do still find this way of coding sometimes quite clear, even if it would likely be changed in the future (perhaps once we do i18n, very likely if we added more information than just the direction).